sky/packages/sky/lib/src/widgets/page.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'navigator2.dart';
+import 'overlay.dart';
+import 'transitions.dart';
+
+abstract class TransitionRoute extends Route {
```

Comment:
Yeah, I think we'll want the Route in navigator.dart to be purely abstract and have a routes.dart that has a bunch of useful route base classes.  I'll handle that when porting over more use cases.